src/snowflake/snowpark/session.py

```diff
@@ -259,10 +259,14 @@ def __init__(self, conn: ServerConnection) -> None:
         self._plan_builder = SnowflakePlanBuilder(self)
         self._last_action_id = 0
         self._last_canceled_id = 0
-        self._use_scoped_temp_objects = (
+        self._use_scoped_temp_objects = bool(
             _use_scoped_temp_objects
-            and conn._conn._session_parameters.get(
-                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
+            and (
+                conn._conn._session_parameters.get(
+                    _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
+                )
+                if conn._conn._session_parameters
+                else True
```

Comment:
We don't have to do it in this PR, but feels like if we have more similar usage, we should create a helper function :)

Reply:
good idea, I could make a comment for this